### ISSUE TITLE
Problems validating user to fetch only his personal data

### ISSUE THREAD

**Issue by dmscn** [diegohaz/rest#120]
I've created an api called "task" restricted to the user who created them. 

When Using Postman to test if it's working i got a `500 Internal Server Error`

Do i have to pass anything besides the user `access_token`?

This is what appears in the log:

```
2017-11-01T23:55:58.667001+00:00 app[web.1]: [0mGET /tasks?access_token=eyKo0dcciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjU5ZmE0OGRmMGM3ODBiMDAxMmQxYThkMCIsFmlfaCI6MTUwRTU4MDQgOX0.9Tx2Vhsic8UC9fgQZU5E734Em1QfJgHpfNbZ1llohwYM [31m500 [0m11.401 ms - 148[0m
2017-11-01T23:55:58.668154+00:00 app[web.1]: TypeError: Cannot read property 'view' of null
2017-11-01T23:55:58.668155+00:00 app[web.1]:     at model.view (/app/src/api/task/model.js:30:13)
2017-11-01T23:55:58.668156+00:00 app[web.1]:     at /app/src/api/task/controller.js:14:47
2017-11-01T23:55:58.668156+00:00 app[web.1]:     at Array.map (native)
2017-11-01T23:55:58.668157+00:00 app[web.1]:     at /app/src/api/task/controller.js:14:28
2017-11-01T23:55:58.668158+00:00 app[web.1]:     at tryCatcher (/app/node_modules/bluebird/js/release/util.js:16:23)
2017-11-01T23:55:58.668159+00:00 app[web.1]:     at Promise._settlePromiseFromHandler (/app/node_modules/bluebird/js/release/promise.js:512:31)
2017-11-01T23:55:58.668160+00:00 app[web.1]:     at Promise._settlePromise (/app/node_modules/bluebird/js/release/promise.js:569:18)
2017-11-01T23:55:58.668160+00:00 app[web.1]:     at Promise._settlePromise0 (/app/node_modules/bluebird/js/release/promise.js:614:10)
2017-11-01T23:55:58.668161+00:00 app[web.1]:     at Promise._settlePromises (/app/node_modules/bluebird/js/release/promise.js:693:18)
2017-11-01T23:55:58.668161+00:00 app[web.1]:     at Async._drainQueue (/app/node_modules/bluebird/js/release/async.js:133:16)
2017-11-01T23:55:58.668162+00:00 app[web.1]:     at Async._drainQueues (/app/node_modules/bluebird/js/release/async.js:143:10)
2017-11-01T23:55:58.668162+00:00 app[web.1]:     at Immediate.Async.drainQueues (/app/node_modules/bluebird/js/release/async.js:17:14)
2017-11-01T23:55:58.668162+00:00 app[web.1]:     at runCallback (timers.js:672:20)
2017-11-01T23:55:58.668163+00:00 app[web.1]:     at tryOnImmediate (timers.js:645:5)
2017-11-01T23:55:58.668163+00:00 app[web.1]:     at processImmediate [as _immediateCallback] (timers.js:617:5)
```

**Comment by diegohaz** [comment]
Could you paste the content of `/app/src/api/task/model.js` and `/app/src/api/task/controller.js`?

**Comment by dmscn** [comment]
`/app/src/api/task/controller.js`
```
import _ from 'lodash'
import { success, notFound, authorOrAdmin } from '../../services/response/'
import { Task } from '.'

export const create = ({ user, bodymen: { body } }, res, next) =>
  Task.create({ ...body, user })
    .then((task) => task.view(true))
    .then(success(res, 201))
    .catch(next)

export const index = ({ querymen: { query, select, cursor } }, res, next) =>
  Task.find(query, select, cursor)
    .populate('user')
    .then((task) => task,.map((task) => task.view()))
    .then(success(res))
    .catch(next)

export const show = ({ params }, res, next) =>
  Task.findById(params.id)
    .populate('user')
    .then(notFound(res))
    .then((task) => task ? task.view() : null)
    .then(success(res))
    .catch(next)

export const update = ({ user, bodymen: { body }, params }, res, next) =>
  Task.findById(params.id)
    .populate('user')
    .then(notFound(res))
    .then(authorOrAdmin(res, user, 'user'))
    .then((task) => task ? _.merge(task, body).save() : null)
    .then((task) => task ? task.view(true) : null)
    .then(success(res))
    .catch(next)

export const destroy = ({ user, params }, res, next) =>
  Task.findById(params.id)
    .then(notFound(res))
    .then(authorOrAdmin(res, user, 'user'))
    .then((task) => task ? task.remove() : null)
    .then(success(res, 204))
    .catch(next)
```

`/app/src/api/task/model.js`

```
import mongoose, { Schema } from 'mongoose'

const taskSchema = new Schema({
  user: {
    type: Schema.ObjectId,
    ref: 'User',
    required: true
  },
  title: {
    type: String
  },
  label: {
    type: String
  },
  date: {
    type: String
  },
  text: {
    type: String
  }
}, {
  timestamps: true
})

taskSchema.methods = {
  view (full) {
    const view = {
      // simple view
      id: this.id,
      user: this.user.view(full),
      title: this.title,
      label: this.label,
      date: this.date,
      text: this.text,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt
    }

    return full ? {
      ...view
      // add properties for a full view
    } : view
  }
}

const model = mongoose.model('Task', taskSchema)

export const schema = model.schema
export default model
```

**Comment by diegohaz** [comment]
```js
export const index = ({ querymen: { query, select, cursor } }, res, next) =>
  Task.find(query, select, cursor)
    .populate('user')
    .then((task) => task,.map((task) => task.view())) <----------- remove comma after second task
    .then(success(res))
    .catch(next)
```

**Comment by dmscn** [comment]
I implemented this like it was said by @DanMMX in #79 and it worked. Just added user to query.
```
export const index = ({ querymen: { query, select, cursor }, user }, res, next) =>
  Task.find({ user: user.id, ...query }, select, cursor)
    .populate('user')
    .then((task) => task.map((task) => task.view()))
    .then(success(res))
    .catch(next)
```

**Comment by diegohaz** [comment]
Alright, glad you've found it.
